src/AutomaticDiContainer.php

```diff
@@ -1,47 +1,61 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Midnight\AutomaticDi;
 
-use Interop\Container\ContainerInterface;
 use LogicException;
+use Psr\Container\ContainerInterface;
 use ReflectionClass;
 use ReflectionParameter;
 
+use function class_exists;
+use function interface_exists;
+use function sprintf;
+
 class AutomaticDiContainer implements ContainerInterface
 {
-    /** @var ContainerInterface */
-    private $container;
-    /** @var AutomaticDiConfig */
-    private $config;
+    private ContainerInterface $container;
+    private AutomaticDiConfig $config;
 
     public function __construct(ContainerInterface $container, AutomaticDiConfig $config)
     {
         $this->container = $container;
         $this->config = $config;
     }
 
+    /**
+     * @param string $id
+     * @return mixed|object
```

Comment:
`mixed` is a supertype of `object`.
```suggestion
     * @return mixed
```